### PR TITLE
As a user, when I export data OSIS should always be the far left column.

### PR DESCRIPTION
**Pull request by yasserhussain1110**
OSIS  should now show up on the left hand side always.

### CODE REVIEW

File: app/client/src/components/studentFilter/exportFields.js [hunk 1]
```diff
@@ -103,19 +109,16 @@ class ExportCSV extends Component {
...
+        const selectedKeys = addFieldOSISToTheLeft(
+            // Get all keys in the studentCategoryObj
+            Object.keys(studentCategoryObj)
+                // Filter out those keys which do not have a truthy value in component's state
+                .filter(key => !!this.state[key])
+                // Get the values corresponding to those keys in studentCategoryObj
+                .map(key => studentCategoryObj[key])
+                // Concat all values to get a list of fields
+                .reduce((a, b) => a.concat(b), [])
         );
```
Comment (sachinmour) [inline app/client/src/components/studentFilter/exportFields.js]: You are looping over the keys three times un-necessarily.. please explain what's the problem in the old code

File: app/client/src/components/studentFilter/exportFields.js [hunk 2]
```diff
@@ -19,6 +19,12 @@ _.forOwn(studentCategoryObj, (value, key) => {
     studentCategoryObj[key] = studentCategoryObj[key].map(field => field.dbName);
 });
 
+const addFieldOSISToTheLeft = keys => {
+    let osisIndex = keys.indexOf('osis');
```
Comment (sachinmour) [inline app/client/src/components/studentFilter/exportFields.js]: look into _.remove.. way easier to use and can be used directly when calculating selectedKeys down in handleStudents.
eg. const selectedKeys = _(studentCategoryObj).reduce(......).remove(key => key === 'osis');
selectedKeys.unshift('osis');

Comment (yasserhussain1110) [inline app/client/src/components/studentFilter/exportFields.js]: `remove` seems to mutate the array and return the removed keys.
So, `selectedKeys` will have the removed keys, isn't it?
How will it work in this situation?

Comment (sachinmour) [inline app/client/src/components/studentFilter/exportFields.js]: i updated the comment

File: app/client/src/components/studentFilter/exportFields.js (hunk 1 above)
Comment (yasserhussain1110) [inline app/client/src/components/studentFilter/exportFields.js]: This was much more readable to me. The performance change is almost negligible.

File: app/client/src/components/studentFilter/exportFields.js (hunk 2 above)
Comment (yasserhussain1110) [inline app/client/src/components/studentFilter/exportFields.js]: This code should work, but my question is - Doesn't it cause confusion? Why not just use `filter`?

Comment (yasserhussain1110) [inline app/client/src/components/studentFilter/exportFields.js]: This code should work, but my question is - Doesn't it cause confusion? Why not just use filter?

**Pull request closed without merging**